### PR TITLE
Overview openapi

### PR DESCRIPTION
**Pull request by artemkovalyov**
## What Has Changed?

OpenAPI overview

## Manual Checks?

- [x] Text adheres to the style guide (`vale docs/`)
  - Every sentence is on its own line
  - Headings use [title capitalization](https://capitalizemytitle.com/style/AP/#) (applies also to the sidebar and title of a document)
  - You followed [naming center](https://www.sapbrandtools.com/naming-center/#/dashboard) guidelines when referring to SAP products (e.g. SAP S/4HANA)
- [x] You checked your spelling and grammar (consider using Grammarly, see `CONTRIBUTING.md`)
- [x] You formatted all changed files with prettier (`npm run prettier`)
- [x] You tested if the documentation still builds (`npm run build`)
- [x] You verified all new and changed links still work (changing the `id` or name of a file can break links)
- [x] You have updated the [feature matrix](https://github.com/SAP/cloud-sdk/blob/main/docs/components/data/features.js) if you add documentation on a new feature or otherwise required.

### CODE REVIEW

File: docs/java/features/rest/overview.mdx [hunk 1]
```diff
@@ -4,43 +4,48 @@ title: Overview
...
 
-In case you need information on REST or other services shipped only internally, please, approach us directly via standard communication channels.
+Depending on the scope, modules could be available only internally within SAP or publicly released.
+The [SAP Workflow service](clients/scp-workflow-rest-api) is an example of a publicly released API module.
 
 ## Why Using SAP Cloud SDK for Java With OpenAPI Services?
 
-- You'll benefit from a type-safe client when accessing a service of your choice
-- We take care of various complexities around developing of applications and extensions on SAP Business Technology Platform.
+- You'll benefit from a type-safe, less code boilerplate, and better developer experience.
```
Comment (sacnl) [inline docs/java/features/rest/overview.mdx:44]:
```suggestion
- You'll benefit from type-safety, less code boilerplate, and better developer experience.
```

File: docs/java/features/rest/overview.mdx [hunk 2]
```diff
@@ -4,43 +4,48 @@ title: Overview
...
+[OpenAPI](https://www.openapis.org/) standard is a popular format to define RESTful APIs. Together with [OData](../odata/overview.mdx) it is one of the key API definition formats used at SAP. The SAP Cloud SDK provides convenient tooling to make the consumption of OpenAPI services easy and convenient. We provide a customized [OpenAPI code generator](generate-rest-client.mdx) capable of converting an OpenAPI specification into a type-safe Java client library nicely integrated with the SAP Business Technology Platform abstractions also provided by the SAP Cloud SDK. To speed up things for developers and reduce maintenance, we are working with different SAP services to release pre-generated client libraries available through [Maven Central](https://search.maven.org/search?q=g:com.sap.cloud.sdk*). Another way to get a client library is to generate it based on API specifications available at [SAP API Business Hub](https://api.sap.com/). If you encounter any problems or errors with OpenAPI tooling give us feedback via our [support channels](../../../overview/support.mdx).
```
Comment (sacnl) [inline docs/java/features/rest/overview.mdx]: "SAP Business Technology Platform abstractions also provided by the SAP Cloud SDK" - can you add a link here?
Maybe you want to make even clearer that the pre-generated libraries are an additional convenience, but not necessary for developers to benefit from the generator.

**Review by sacnl — COMMENTED** [review]
Thanks

File: docs/java/features/rest/overview.mdx (hunk 2 above)
Comment (artemkovalyov) [reply docs/java/features/rest/overview.mdx]: @sacnl, I think it has enough links. A customers can read about our other features by navigating through the docs. There they are explained with enough details.

File: docs/java/features/rest/overview.mdx [hunk 3]
```diff
@@ -4,43 +4,56 @@ title: Overview
...
+:::
+
 ## OpenAPI Based Type-Safe Client for RESTful Services in Java
 
-The SAP Cloud SDK for Java is prepared to generate type-safe clients using [OpenAPI](https://www.openapis.org/) to conveniently work with various OpenAPI based services available at [SAP API Hub](https://api.sap.com/).
+[OpenAPI](https://www.openapis.org/) standard is a popular format to define RESTful APIs.
+Together with [OData](../odata/overview.mdx) it is one of the key API definition formats used at SAP.
+The SAP Cloud SDK provides convenient tooling to make the consumption of OpenAPI services easy and convenient.
+We provide a customized [OpenAPI code generator](generate-rest-client.mdx) capable of converting an OpenAPI specification into a type-safe Java client library nicely integrated with the SAP Business Technology Platform abstractions also provided by the SAP Cloud SDK.
+To speed up things for developers and reduce maintenance, we are working with different SAP services to release pre-generated client libraries available through [Maven Central](https://search.maven.org/search?q=g:com.sap.cloud.sdk*).
```
Comment (sacnl) [inline docs/java/features/rest/overview.mdx:31]:
```suggestion
On top, for additional convenience, to speed up things for developers and reduce maintenance, we are working with different SAP services to release pre-generated client libraries available through [Maven Central](https://search.maven.org/search?q=g:com.sap.cloud.sdk*).
```

File: docs/java/features/rest/overview.mdx [hunk 4]
```diff
@@ -4,43 +4,56 @@ title: Overview
...
+Another way to get a client library is to generate it based on API specifications available at [SAP API Business Hub](https://api.sap.com/).
```
Comment (sacnl) [inline docs/java/features/rest/overview.mdx:32]:
```suggestion
Another way to get a client library is to generate it yourself based on API specifications available at [SAP API Business Hub](https://api.sap.com/).
```

File: docs/java/features/rest/overview.mdx [hunk 5]
```diff
@@ -4,43 +4,56 @@ title: Overview
...
+
+:::
+
 ## OpenAPI Based Type-Safe Client for RESTful Services in Java
 
-The SAP Cloud SDK for Java is prepared to generate type-safe clients using [OpenAPI](https://www.openapis.org/) to conveniently work with various OpenAPI based services available at [SAP API Hub](https://api.sap.com/).
+[OpenAPI](https://www.openapis.org/) standard is a popular format to define RESTful APIs.
+Together with [OData](../odata/overview.mdx) it is one of the key API definition formats used at SAP.
+The SAP Cloud SDK provides convenient tooling to make the consumption of OpenAPI services easy and convenient.
+We provide a customized [OpenAPI code generator](generate-rest-client.mdx) capable of converting an OpenAPI specification into a type-safe Java client library nicely integrated with the SAP Business Technology Platform abstractions also provided by the SAP Cloud SDK.
```
Comment (sacnl) [inline docs/java/features/rest/overview.mdx:30]:
```suggestion
We provide a customized [OpenAPI code generator](generate-rest-client.mdx) capable of converting an OpenAPI specification into a type-safe Java client library nicely integrated with the SAP Business Technology Platform abstractions also provided by the SAP Cloud SDK.
You can use that generator to create a client for an OpenAPI-based API of your choice.
```

**Review by sacnl — COMMENTED** [review]
A few suggestions

File: docs/java/features/rest/overview.mdx (hunk 5 above)
Comment (artemkovalyov) [reply docs/java/features/rest/overview.mdx:30]: Is it smth special to our policies: "of your choice"? I meet it often but from a meaning perspective, it seems rather verbose if not to say redundant. No hard feelings, I'll accept it. I personally stumble over those and it makes docs more difficult to read for me.

File: docs/java/features/rest/overview.mdx (hunk 4 above)
Comment (artemkovalyov) [reply docs/java/features/rest/overview.mdx:33]: This also seems self-explanatory to me. Who else would generate it for you? But if you've seen our customers missing the point in such situation I'll accept it.

File: docs/java/features/rest/overview.mdx (hunk 3 above)
Comment (artemkovalyov) [reply docs/java/features/rest/overview.mdx:32]: Here I'd leave either my version or yours. If yous to be taken it's better to remove "to speed up things for developers and reduce maintenance" as yours already implies it. The main goal of this abstract is still emphasizing clients at the end.

**Comment by artemkovalyov** [comment]
@sacnl, let's decide about the first suggestion and merge it soon. Then I can refer here from the blog and start sharing it.

**Comment by sacnl** [comment]
All of my comments were suggestions to bring across the point very clearly that there is a difference between generator and pre-generated, and that already the first offers value. I have seen users stumble over that difference already with OData. No hard feelings if you leave out some of the suggestions due to redundancy :)

**Review by sacnl — APPROVED** [review]

**Comment by artemkovalyov** [comment]
Good. I believed otherwise as you requested those as changes:) I really hope our support and more savvy developers + open-sourcing things at some point will help to make our comms easier. I now can see some tools and products from SAP can be very restrictive which is obviously much less of a case with the SDK.

**Pull request merged**